Srcs/heiftojpeg/heiftojpeg.cpp

```diff
@@ -0,0 +1,227 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <ctype.h>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <Magick++.h>
+#include <list>
+#include "hevcimagefilereader.hpp"
+
+using namespace std;
+
+static int VERBOSE = 0;
+static int MAX_SIZE = -1;
+
+static void decodeData(ImageFileReaderInterface::DataVector data, Magick::Image *image)
+{
+    string hevcFileName = tmpnam(nullptr);
+    string bmpFileName = tmpnam(nullptr);
+    bmpFileName += ".bmp"; // ffmpeg expects the output file to the have the extension ".bmp"
+    ofstream hevcFile(hevcFileName);
+    if (!hevcFile.is_open()) {
+        cerr << "could not open " << hevcFileName << " for writing HEVC\n";
+        exit(1);
+    }
+    hevcFile.write((char*)&data[0], data.size());
+    if (hevcFile.bad()) {
+        cerr << "failed to write " << data.size() << " bytes of HEVC to " << hevcFileName << "\n";
+        exit(1);
+    }
+    hevcFile.close();
+    if (VERBOSE) {
+        cout << "wrote " << data.size() << " bytes of HEVC to " << hevcFileName << "\n";
+    }
+    int retval = system(("ffmpeg -i " + hevcFileName + " -loglevel panic -frames:v 1 -vsync vfr -q:v 1 -y -an " + bmpFileName).c_str());
```

Comment:
it would be nice to add timings (here's some block foo to get std::chrono::milliseconds https://stackoverflow.com/a/21995693/1451434) to know how long this tile decoding takes on our c4.large and what portion of the total process it is.  montage/zoom/jpeg of full image probably dominates?  if this decoding is a big part of the time, we should probably run it under "parallel --gnu" or posix_spawn it ourselves since we're already paying the 4096*3072*3=36MB memory footprint of the full image during the montage, just like we do when we rotate other images.

Reply:
added

Reply:
I'm going to move towards merging and deploying this thing. After it's been deployed I can test it with `-v` and get the timing data on real hosts in the containers.  On the `m1.medium` controller host it takes ~10s to process an image and resize to 2048w. About half of that is serialized 100ms calls to ffmpeg, the other half is about 2s to zoom, 2s to montage and write the JPEG, and 1s to call exiftool. The image magick parts are significantly faster on my laptop, but the ffmpeg calls are similar.

Reply:
actually c4.large only have 2 cores so we'd probably have to upgrade the instances if we really wanted to make this lower latency.  and ffmpeg may be decoding hevc with multiple threads already but i'm not sure what's really dominating anyways without us digging in deeper.  using /run/shm for the temp files didn't make any difference

Reply:
on a staging t2.medium, it completes the job in under 4s. each tile takes 40ms to process with ffmpeg, and the montage/resize takes only 1.5s. exiftool takes 400ms.